packages/core/src/indexing/database-index-api.ts

```diff
@@ -29,6 +31,25 @@ export interface IndexStreamArgs {
   readonly firstAnchor: Date | null
 }
 
+/**
+ * Field definition for index
+ */
+export interface Field {
+  path: Array<string>
+}
+
+/**
+ * Index for a model
+ */
+export type ModelIndex = {
+  name: string
```

Comment:
Is the `name` provided by the client please? If so, how should they be defined? What is the behavior if duplicate names are used?